### PR TITLE
Fix hash removal from URL and browser history after interactive request

### PR DESCRIPTION
**Pull request by hectormmg**
This PR:

- Refactors `BrowserUtils.clearHash` to accept a window as a parameter and perform hash removal/history replacement on that content window.
- Fixes a bug in which the auth code hash was being left in the browser history

### CODE REVIEW

**Review by jasonnutter — APPROVED** [review]

**Review by tnorling — APPROVED** [review]

**Review by jo-arroyo — APPROVED** [review]

**Review by jasonnutter — CHANGES_REQUESTED** [review]
Marking as requesting changes until updates are pushed.

**Comment by hectormmg** [comment]
@jasonnutter should be gtm now, sorry

File: lib/msal-angular/src/msal.navigation.client.ts [hunk 1]
```diff
@@ -34,8 +34,12 @@ export class MsalCustomNavigationClient extends NavigationClient {
 
         // Replaces current state if noHistory flag set to true
         this.authService.getLogger().verbosePii(`MsalCustomNavigationClient - navigating to newUrl: ${newUrl}`);
-        this.router.navigateByUrl(newUrl, { replaceUrl: options.noHistory });
 
-        return Promise.resolve(false);
+        if (options.noHistory) {
```
[the remark this answers is not in the document]
Comment (hectormmg) [reply lib/msal-angular/src/msal.navigation.client.ts:38]:
```suggestion
        // Prevent hash clearing from causing an issue with Client-side navigation after redirect is handled
        if (options.noHistory) {
```

[the remark this answers is not in the document]
Comment (hectormmg) [reply lib/msal-angular/src/msal.navigation.client.ts:38]: @jasonnutter is this accurate?

**Review by jasonnutter — APPROVED** [review]

**Comment by MaartenX** [comment]
I noticed an extra history entry since this change; The following code is now always executed, even if ```replaceState``` exists:

``` javascript
contentWindow.location.hash = Constants.EMPTY_STRING;
```

Shouldn't the code above only be executed when ```replaceState``` does not exist?

**Comment by WellyHong** [comment]
As @MaartenX  mention above, I also encounter the unexpected behavior after login, it redirect to the default page even if I set `navigateToLoginRequestUrl: false`.

**Pull request merged**